arvo/eyre/data-types.md

```diff
@@ -0,0 +1,290 @@
++$  binding
+  $:  site=(unit @t)
+      path=(list @t)
+  ==
+```
+
+A `binding` is a rule to match a URL `path` and optional `site` domain which can then be tied to an [$action](#action). A `path` of `/foo` will also match `/foo/bar`, `/foo/bar/baz`, etc. If the `site` is `~` it will be determined implicitly. A binding must be unique.
+
+## `$action`
+
+```hoon
++$  action
+  $%  [%gen =generator]    ::  dispatch to a generator
+      [%app app=term]      ::  dispatch to an application
+      [%authentication ~]  ::  internal authentication page
+      [%logout ~]          ::  internal logout page
+      [%channel ~]         ::  gall channel system
+      [%scry ~]            ::  gall scry endpoint
+      [%four-oh-four ~]    ::  respond with the default file not found page
+  ==
+```
+
+The action to take when a [$binding](#binding) matches an incoming HTTP request.
+
+## `$generator`
+
+```hoon
++$  generator
+  $:  =desk           ::  desk on current ship that contains the generator
+      path=(list @t)  ::  path on :desk to the generator's hoon file
+      args=*          ::  args: arguments passed to the gate
+  ==
+```
+
+This refers to a generator on a local ship that can handle requests. Note that serving generators via Eyre is not fully implmented and should not be used.
+
+## `$http-config`
+
+```hoon
++$  http-config
+  $:  secure=(unit [key=wain cert=wain])
+      proxy=_|
+      log=?
+      redirect=?
+  ==
+```
+
+The configuration of the runtime HTTP server itself. The `secure` field contains the PEM-encoded RSA private key and certificate or certificate chain if using HTTPS, otherwise is `~` for plain HTTP. The `proxy` field is not currently used. The `log` field turns on HTTP(s) access logs but is not currently implemented. The `redirect` field turns on 301 redirects to upgrade HTTP to HTTPS if the `key` and `cert` are set in `secure`.
+
+## `$http-rule`
+
+```hoon
++$  http-rule
+  $%  [%cert cert=(unit [key=wain cert=wain])]
+      [%turf action=?(%put %del) =turf]
+  ==
+```
+
+This is for updating the server configuration. In the case of `%cert`, a `cert` of `~` clears the HTTPS cert & key, otherwise `cert` contains the PEM-encoded RSA private key and certificate or certificate chain. In the case of `%turf`, a `%put` `action` sets a domain name and a `%del` `action` removes it. The [$turf](#turf) contains the domain name.
+
+## `$address`
+
+```hoon
++$  address
+  $%  [%ipv4 @if]
+      [%ipv6 @is]
+  ==
+```
+
+A client IP address.
+
+## `$inbound-request`
+
+```hoon
++$  inbound-request
+  $:  authenticated=?
+      secure=?
+      =address
+      =request:http
+  ==
+```
+
+An inbound HTTP request and metadata - this is what is poked into a bound app. The `authenticated` field says whether the request was made with a valid session cookie. The `secure` field says whether it was made with HTTPS. The [$address](#address) is the client's IP address. The [$request:http](#request-http) contains the HTTP request itself. 
```

Comment:
Here and elsewhere, I'm not sure how common "poke into" is in the rest of the docs. It feels a little off to me, compared to the slightly more verbose "app is poked with ...". No strong feelings, but want to offer up this alternative for your consideration.